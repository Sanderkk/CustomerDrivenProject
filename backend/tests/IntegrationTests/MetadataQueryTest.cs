using HotChocolate;
using HotChocolate.Execution;
using HotChocolate.Types;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using src.Api.Inputs;
using src.Api.Queries;
using src.Config;
using src.Database;
using src.Api.Types;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using Xunit;

namespace tests.IntegrationTests
{
    public class MetadataQueryTest
    {
        //Database Connect config (move to own config file)
        private string dbConnectionString = "Host=sanderkk.com;Username=sintef;Password=123456;Database=fishfarm2";

        [Fact]
        public async Task AllMetadataQuery()
        {
            //Arrange
            //Connect to database
            IServiceProvider serviceProvider =
                new ServiceCollection()
                .AddSingleton<IFishFarmRepository, FishFarmRepository>()
                .AddSingleton<IDatabaseConfig>(sp =>
                new DatabaseConfig()
                {
                    DatabaseConnectionString = dbConnectionString
                }
                )
                .BuildServiceProvider();

            IQueryExecutor executor = Schema.Create(c =>
            {
                c.RegisterQueryType(new ObjectType<MetadataQuery>(d => d.Name("Query")));
                c.RegisterType<List<MetadataType>>();
            })
            .MakeExecutable();

            IReadOnlyQueryRequest request =
                QueryRequestBuilder.New()
                .SetQuery(@"{allMetadata(sensorID:1){metadataID,sensorID,name,serialNumber,number,department,company,purchaseDate,lending,lendingPrice,cableLength,checkOnInspectionRound,company,voltage,coordinate, altitude, locationDescription, createdAt, updatedAt, outdatedFrom}}")
                .SetServices(serviceProvider)
                .Create();

            // act
            IExecutionResult result = await executor.ExecuteAsync(request);
            //Assert
            Snapshot.Match(result);
        }

        [Fact]
        public async Task LastMetadataQuery()
        {
            //Arrange
            //Connect to database
            IServiceProvider serviceProvider =
                new ServiceCollection()
                .AddSingleton<IFishFarmRepository, FishFarmRepository>()
                .AddSingleton<IDatabaseConfig>(sp =>
                new DatabaseConfig()
                {
                    DatabaseConnectionString = dbConnectionString
                }
                )
                .BuildServiceProvider();

            IQueryExecutor executor = Schema.Create(c =>
            {
                c.RegisterQueryType(new ObjectType<MetadataQuery>(d => d.Name("Query")));
                c.RegisterType<List<MetadataType>>();
            })
            .MakeExecutable();

            IReadOnlyQueryRequest request =
                QueryRequestBuilder.New()
                .SetQuery(@"{lastMetadata(sensorID:1){metadataID,sensorID,name,serialNumber,number,department,company,purchaseDate,lending,lendingPrice,cableLength,checkOnInspectionRound,company,voltage,coordinate, altitude, locationDescription, createdAt, updatedAt, outdatedFrom}}")
                .SetServices(serviceProvider)
                .Create();

            // act
            IExecutionResult result = await executor.ExecuteAsync(request);
            //Assert
            Snapshot.Match(result);
            //result.MatchSnapshot();
        }
    }
}