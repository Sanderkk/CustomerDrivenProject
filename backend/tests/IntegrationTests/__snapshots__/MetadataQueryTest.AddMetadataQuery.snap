{
  "Data": {
    "addMetadata": {
      "sensorID": 5,
      "locationID": 65,
      "name": "Jasmine",
      "serialNumber": "55yh3kk3-4443-455ddw2",
      "number": "55T77",
      "modelNumber": "jjhj445",
      "department": "IDI",
      "company": "kundestyrt and co",
      "ownerID": "33333",
      "purchaseDate": "2002-12-25T00:00:00.000+01:00",
      "lending": true,
      "lendingPrice": 9000.0,
      "cableLength": 125.0,
      "checkOnInspectionRound": false,
      "inspectionRound": "same like the others",
      "servicePartner": "kundestyrt and co",
      "voltage": "24volt",
      "coordinate": "Moholt",
      "altitude": 55,
      "locationDescription": "samsy like always",
      "identificator": "nothing special here",
      "measureArea": "550 550",
      "picture": "Some picture link or svg?",
      "plannedDisposal": "2022-12-25T00:00:00.000+01:00",
      "actualDisposal": "2022-12-25T00:00:00.000+01:00",
      "warrantyDate": "2007-12-25T00:00:00.000+01:00",
      "signal": "Wireless",
      "tag1": "nada",
      "tag2": "tag222222",
      "tag3": "im tagged boi",
      "timeless": false,
      "tollerance": true,
      "outdatedFrom": null
    }
  },
  "Extensions": {},
  "Errors": [],
  "ContextData": {}
}
